boards/temperature/src/communication.rs

```diff
@@ -1,34 +1,76 @@
+use core::num::{NonZeroU16, NonZeroU8};
+
 use crate::data_manager::DataManager;
 use crate::types::COM_ID;
 use common_arm::HydraError;
-use defmt::{error, info};
+use defmt::info;
+use fdcan::Instance;
 use fdcan::{
+    config::NominalBitTiming,
+    filter::{StandardFilter, StandardFilterSlot},
     frame::{FrameFormat, TxFrameHeader},
     id::StandardId,
 };
-use mavlink::peek_reader::PeekReader;
-use messages::mavlink::uorocketry::MavMessage;
-use messages::mavlink::{self};
 use messages::Message;
 use postcard::from_bytes;
 
 /// Clock configuration is out of scope for this builder
 /// easiest way to avoid alloc is to use no generics
-pub struct CanCommandManager {
-    can: fdcan::FdCan<
-        stm32h7xx_hal::can::Can<stm32h7xx_hal::pac::FDCAN1>,
-        fdcan::NormalOperationMode,
-    >,
+pub struct CanManager<I: Instance> {
+    can: fdcan::FdCan<I, fdcan::NormalOperationMode>,
 }
 
-impl CanCommandManager {
-    pub fn new(
-        can: fdcan::FdCan<
-            stm32h7xx_hal::can::Can<stm32h7xx_hal::pac::FDCAN1>,
-            fdcan::NormalOperationMode,
-        >,
-    ) -> Self {
-        Self { can }
+impl<I: Instance> CanManager<I> {
+    pub fn new(mut can: fdcan::FdCan<I, fdcan::ConfigMode>) -> Self {
+        // let data_bit_timing = DataBitTiming {
+        //     prescaler: NonZeroU8::new(10).unwrap(),
+        //     seg1: NonZeroU8::new(13).unwrap(),
+        //     seg2: NonZeroU8::new(2).unwrap(),
+        //     sync_jump_width: NonZeroU8::new(4).unwrap(),
+        //     transceiver_delay_compensation: true,
+        // };
+        // can_data.set_automatic_retransmit(false); // data can be dropped due to its volume.
+
+        // can_command.set_data_bit_timing(data_bit_timing);
+        let btr = NominalBitTiming {
+            prescaler: NonZeroU16::new(10).unwrap(),
+            seg1: NonZeroU8::new(13).unwrap(),
+            seg2: NonZeroU8::new(2).unwrap(),
+            sync_jump_width: NonZeroU8::new(1).unwrap(),
+        };
+
+        // Why?
+        can.set_protocol_exception_handling(false);
```

Comment:
I think this is left over from hydra since I wanted to dump CAN messages as fast as possible and this was a performance improvement.